org.elastic.jdt.ls.core/src/org/elastic/jdt/ls/core/internal/SecurityPermissionHelper.java

```diff
@@ -0,0 +1,115 @@
+			"( java.net.SocketPermission \"localhost\" \"resolve\" )" +
+			"( java.net.NetPermission \"getNetworkInformation\" )" +  
+			"( java.lang.reflect.ReflectPermission \"suppressAccessChecks\" )" +
+			"( java.io.FilePermission \"<<ALL FILES>>\" \"read\" )" + 	
+			"( java.io.FilePermission \"%s\" \"write,delete\" )" + 
+			"( java.io.SerializablePermission \"enableSubstitution\" )" +
+			"( java.lang.RuntimePermission \"*\" \"*\" )" + 
+			"( java.util.PropertyPermission \"*\" \"read, write\" )" +
+			"( java.net.NetPermission \"getProxySelector\" )" +
+			"%s } \"Normal bundles\"";
+	
+	private SecurityPermissionHelper() {}
+	
+	public static void setSecurityPermissions(BundleContext bundleContext) {
+		
+                ServiceReference sRef = bundleContext.getServiceReference(ConditionalPermissionAdmin.class.getName());
+                
+                // Get hold of OSGi ConditionalPermissionAdmin service...
+                ConditionalPermissionAdmin conPermAdmin = (ConditionalPermissionAdmin) bundleContext.getService(sRef);
+		
+                // Create new "atomic rules update" object
+                ConditionalPermissionUpdate update = conPermAdmin.newConditionalPermissionUpdate();
+
+                // Get list of existing permissions (normally null)
+                List<ConditionalPermissionInfo> perms = update.getConditionalPermissionInfos();
+                
+                // Clear old permissions
+                update.getConditionalPermissionInfos().clear();
+                
+                ConditionalPermissionInfo coreP = conPermAdmin.newConditionalPermissionInfo(CORE_PERM);
+                perms.add(coreP);
+                ConditionalPermissionInfo denyP = conPermAdmin.newConditionalPermissionInfo(DENY_EXEC_PERM);
+                perms.add(denyP);
+                String minimalPerm = String.format(
+                                MINIMAL_PERM_TEMPLATE,
+                                Environment.get(CLIENT_HOST) + ":" + Environment.get(CLIENT_PORT),
+                                getDataFolder(),
+                                getSocketPermissions());
+                ConditionalPermissionInfo minimalPermission = conPermAdmin.newConditionalPermissionInfo(minimalPerm);
+                perms.add(minimalPermission);
+	       
+                update.commit();
+	}
+	
+	private static String getDataFolder() {
+		return Platform.getLocation().toFile().getParent() + File.pathSeparator + "-";
+	}
+	
+	private static String getSocketPermissions() {
+		String extraHosts = Environment.getEnvironment(EXTRA_WHITELIST_HOST);
+		List<String> hostsWhiteList = DEFAULT_HOST_WHITELIST;
+		if (extraHosts != null) {
+			hostsWhiteList.addAll(Arrays.asList(extraHosts.split("\\s*,\\s*")));
+		}
+		return String.join(" ", hostsWhiteList.stream().map(host -> String.format("( java.net.SocketPermission \"%s\" \"connect,accept,resolve\" )", host)).collect(Collectors.toList()));
```

Comment:
similar problem, could the white list contain any special character and be used for injection attack?

Reply:
So I need to check if `hostsWhiteList` is a valid host name,  and escape special characters, right?